### PR TITLE
Fixes #1960 Transparent proxy not appending existing X-Forwarded-For header

### LINKED ISSUE #1960: Transparent proxy not appending existing X-Forwarded-For header
### 1. What version of Caddy are you using (`caddy -version`)?
Caddy 0.10.10 (non-commercial use only)

### 2. What are you trying to do?
Use Caddy as a transparent proxy, passing all the IP addresses along the proxy chain to the web app.

### 3. What is your entire Caddyfile?
```text
https://example.net {
        log     syslog
        errors  syslog
        gzip
        proxy   /        127.0.0.1:8080 {
                transparent
        }
}
```

### 4. How did you run Caddy (give the full command and describe the execution environment)?

Caddy is started with the systemd service.

### 5. Please paste any relevant HTTP request(s) here.

Execute this on machine `3.3.3.3`
```
curl -H "X-Forwarded-For: 1.1.1.1,2.2.2.2" https://example.net/print-headers
```

### 6. What did you expect to see?

The actual web app should receive:
```
X-Forwarded-For: 1.1.1.1,2.2.2.2,3.3.3.3
```

### 7. What did you see instead (give full error messages and/or log)?

The web app received instead:
```
X-Forwarded-For: 3.3.3.3
```

### 8. How can someone who is starting from scratch reproduce the bug as minimally as possible?

1. Set up a simple web server that logs HTTP headers on port 8080 (actually a simple `netcat -lp8080` will do).
2. HTTPS should not affect this issue, so change the Caddyfile to use HTTP only
3. Replace `example.net` with `localhost`

### PR DESCRIPTION
**Pull request by theodesp**
### 1. What does this change do, exactly?
It fixes #1960 and adds some tests to verify.

### 2. Please link to the relevant issues.
#1960

### 3. Which documentation changes (if any) need to be made because of this PR?
We may need to mention the transparent preset already appends any **X-Forwarded-For** headers

### 4. Checklist

- [x] I have written tests and verified that they fail without my change
- [ ] I have squashed any insignificant commits
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I am willing to help maintain this change if there are issues with it later

### CODE REVIEW

File: caddyhttp/proxy/proxy_test.go [hunk 1]
```diff
@@ -913,6 +913,68 @@ func TestHostSimpleProxyNoHeaderForward(t *testing.T) {
...
 
+func TestReverseProxyTransparentHeaders(t *testing.T) {
+	testCases := []struct{
+		name string
+		remoteAddr string
+		forwardedForHeader string
+		expected []string
+	} {
+		{"No header", "192.168.0.1:80", "", []string{"192.168.0.1"}},
+		{"Existing", "192.168.0.1:80", "1.1.1.1, 2.2.2.2", []string{"1.1.1.1, 2.2.2.2, 192.168.0.1"}},
```
Comment (theodesp) [inline caddyhttp/proxy/proxy_test.go]: 2 Test cases. One for no existing **X-Forwarded-For** header and one with existing

File: caddyhttp/proxy/proxy_test.go [hunk 2]
```diff
@@ -943,11 +1005,23 @@ func TestHostHeaderReplacedUsingForward(t *testing.T) {
 }
 
 func TestBasicAuth(t *testing.T) {
-	basicAuthTestcase(t, nil, nil)
-	basicAuthTestcase(t, nil, url.UserPassword("username", "password"))
-	basicAuthTestcase(t, url.UserPassword("usename", "password"), nil)
-	basicAuthTestcase(t, url.UserPassword("unused", "unused"),
-		url.UserPassword("username", "password"))
+	testCases := []struct{
```
Comment (theodesp) [inline caddyhttp/proxy/proxy_test.go]: Rewritten this test case into a sub-test case style for better readability

File: caddyhttp/proxy/upstream.go [hunk 3]
```diff
@@ -431,9 +431,10 @@ func parseBlock(c *caddyfile.Dispenser, u *staticUpstream, hasSrv bool) error {
 		}
 		u.downstreamHeaders.Add(header, value)
 	case "transparent":
+		// Note: X-Forwarded-For header is always being appended for proxy connections
```
Comment (theodesp) [inline caddyhttp/proxy/upstream.go]: Added a comment to reason about why the **X-Forwarded-For** header is not set here

File: caddyhttp/proxy/upstream_test.go [hunk 4]
```diff
@@ -316,6 +317,10 @@ func TestParseBlock(t *testing.T) {
 			if _, ok := headers["X-Forwarded-Proto"]; !ok {
 				t.Errorf("Test %d: Could not find the X-Forwarded-Proto header", i+1)
 			}
+
+			if _, ok := headers["X-Forwarded-For"]; ok {
```
Comment (theodesp) [inline caddyhttp/proxy/upstream_test.go]: Test that we do not set the **X-Forwarded-For** when using the transparent Preset

**Comment by theodesp** [comment]
It looks like some formatting errors are in place. Will fix today

**Comment by theodesp** [comment]
Yes, that's one of the merits of Production ready software. 🤓🤓🤓

**Comment by mholt** [comment]
Nice! Thanks, @theodesp! Really appreciate your contribution. :) The test looks really great too.

I'll send you an invite for collaborator privileges so you can help review and merge PRs, triage issues, etc, or whatever else floats your boat. Thanks for being a part of our community!

**Pull request merged**